sound/soc/sof/sof-audio.h

```diff
@@ -79,6 +79,8 @@ struct snd_sof_control {
 	struct list_head list;	/* list in sdev control list */
 
 	struct snd_sof_led_control led_ctl;
+
+	bool updated_in_firmware;
```

Comment:
does this mean volatile?

I am struggling to understand what this means in terms of access.

Reply:
Not volatile, but the value/data of the control can change in SOF and the kernel should sync it's cached value to reflect the current state.

Reply:
this flag would make the kcontrol get behaviour equivalent to a volatile get. I'd like a different name more indicative name something like "control_data_dirty" maybe?

Reply:
In a way, yes, it makes it to do an IPC like the volatile does if the flag is set, but this flag will not make the control similar to the volatile type we currently have.
If the firmware does not send event for a control (ever) then this flag will not change the behavior.

I would choose something different than `control_data_dirty`, the control have value(s) or data, the code is specific about that.
A simple `dirty` perhaps? With a comment that the firmware's and kernel's value/data is out of sync and need to be refreshed. Or perhaps `changed_in_firmware` or `ipc_refresh` ?

Reply:
just dirty seems too simple. elem_value_dirty? or simple elem_dirty?

Reply:
I'll name it after the IPC wrapper function we use to get the value/data: `snd_sof_ipc_set_get_comp_data()`
Let's try: `comp_data_dirty` ?